File: js/scraping.js
const fs = require ('fs')
const {Builder, By} = require ('selenium-webdriver');

async function scrapeBooks() {
    // Initialise webdriver
    let driver = await new Builder().forBrowser('chrome').build();

    try {
        await driver.get('http://books.toscrape.com/');

        // Arrays to store titles and prices
        let titles = [];
        let prices = [];

        // Loop through the first 2 pages
        for (let page = 1; page <= 2; page++) {
            // Wait for the page to load
            await driver.sleep(2000);

            let books = await driver.findElements(By.css('article.product_pod'));

            // Extract title and price for each book
            for (let book of books) {
                let titleElement = await book.findElement(By.css('h3 > a'));
                let title = await titleElement.getAttribute('title');

                let priceElement = await book.findElement(By.css('p.price_color'));
                let price = await priceElement.getText();

                titles.push(title);
                prices.push(price);
            }

            // Navigate to next page if not on the last page
            if (page < 2) {
                let nextButton = await driver.findElement(By.css('li.next > a'));
                await nextButton.click()
            }
        }

        // Print the results
        for (let i = 0; i < titles.length; i++) {
            //console.log('Title: ' + titles[i] + ', Price: ' + prices[i]);
            console.log(`Title: ${titles[i]}, Price: ${prices[i]}`);
        }

        // Save the results to a csv file
        let csvContent = 'Title,Price\n';
        for (let i = 0; i < titles.length; i++) {
            csvContent += `"${titles[i]}","${prices[i]}"\n`;
        }
        fs.writeFileSync('books.csv', csvContent)

    }
    catch(error) {
        console.error(error);
    } finally {
        // Ensure the driver quits
        await driver.quit();

    }
    
}

scrapeBooks()

File: py/scraping.py
from selenium import webdriver
from selenium.webdriver.common.by import By
import time
import csv

def scrapeBooks():

    try:
        driver = webdriver.Edge()
        driver.get('http://books.toscrape.com')

        titles = []
        prices = []

        for page in range(2):
            time.sleep(1)
            books = driver.find_elements(By.CSS_SELECTOR, 'article.product_pod')
            #books = driver.find_elements(By.CLASS_NAME, 'article.product_pod')

            print (page, len(books))

            for book in books:
                titleElement = book.find_element(By.CSS_SELECTOR, 'h3 > a')
                title = titleElement.get_attribute('title')

                priceElement = book.find_element(By.CSS_SELECTOR, 'p.price_color')
                price = priceElement.text

                titles.append(title)
                prices.append(price)

            if page < 2:
                nextButton = driver.find_element(By.CSS_SELECTOR, 'li.next > a')
                #nextButton = driver.find_element(By.CLASS_NAME, 'next')
                nextButton.click()

            csvrowsdata = []
            for t, p in zip (titles, prices):
                #print (t, p)
                csvrowsdata.append({"Title":t, "Price":p})
            
            rows = zip(titles, prices)

            #for i in range(len(books)):
            #    print('Title: {}, Price: {}'.format(titles[i], prices[i]))
            
            with open("books.csv", "w", newline='') as csvile:
                writer = csv.DictWriter(csvile, ["Title", "Price"], restval='')
                writer.writeheader()
                writer.writerows(csvrowsdata)
            
            # with open("books.csv", "w", newline='') as f:
            #     writer = csv.writer(f)
            #     for row in rows:
            #         writer.writerow(row)
    
    except Exception as ex:
        print (ex)
        

scrapeBooks()